File: public/docs/_examples/pipes/ts/src/app/power-boost-calculator.ts
// #docregion
import {Component, FORM_DIRECTIVES} from 'angular2/core'
import {ExponentialStrengthPipe} from './exponential-strength-pipe'

@Component({
  selector: 'power-boost-calculator',
  template: `
    <h2>Power Boost Calculator</h2>
    <div>Normal power: <input [(ng-model)]="power"></div>
    <div>Boost factor: <input [(ng-model)]="factor"></div>
    <p>
      Super Hero Power: {{power | exponentialStrength: factor}}
    </p>
  `,
  pipes: [ExponentialStrengthPipe],
  directives: [FORM_DIRECTIVES]
})
export class PowerBoostCalculator {
  power = 5;
  factor = 1;
}

File: public/docs/_examples/pipes/dart/lib/power_boost_calculator.dart
library pipe_examples.power_boost_calculator;

import 'package:angular2/angular2.dart';
import 'package:pipe_examples/exponential_strength_pipe.dart';

@Component(selector: 'power-boost-calculator')
@View(
    template: '''
<h2>Power Boost Calculator</h2>
<div>Normal power: <input [(ng-model)]="power"></div>
<div>Boost factor: <input [(ng-model)]="factor"></div>
<p>
  Super Hero Power: {{power | exponentialStrength: factor}}
</p>
''',
    pipes: const [ExponentialStrengthPipe],
    directives: const [FORM_DIRECTIVES])
class PowerBoostCalculator {
  int power = 5;
  int factor = 1;
}
